db/models/Entry.js

```diff
@@ -5,6 +5,8 @@ const { Schema } = mongoose;
 const entrySchema = new Schema(
   {
     userId: { type: [Schema.Types.ObjectId], ref: "User" },
+    category: { type: [Schema.Types.ObjectId], ref: "Category" },
```

Comment:
right now the type of userId and category is an `array of mongoDB Ids`. But I think you only want a single ObjectId for the user and the category, right? Then you need to remove the brackets around the Schema.Types.ObjectId